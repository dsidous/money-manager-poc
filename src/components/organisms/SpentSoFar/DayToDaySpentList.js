import React from 'react';
import { View, StyleSheet } from 'react-native';
import { StyledText } from '../../atoms';

const styles = StyleSheet.create({
  container: {
    borderBottomWidth: 1,
    borderBottomColor: '#ccc',
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingVertical: 20,
  },
  title: {
    fontSize: 23,
  },
  amount: {
    fontSize: 24,
  }
});

const DayToDaySpentList = ({ data, sliceColor }) => {
  return data.map((item, i) => (
    <View key={i} style={styles.container}>
      <View style={{ flexDirection: 'row', alignItems: 'center' }}>
        <StyledText style={{ color: sliceColor[i], marginRight: 10 }}>&#9679;</StyledText>
        <StyledText style={styles.title}>{item.category}</StyledText>
      </View>
      <StyledText style={styles.amount}>{`Â£${item.amount}`}</StyledText>
    </View>
  ));
};

export default DayToDaySpentList;
