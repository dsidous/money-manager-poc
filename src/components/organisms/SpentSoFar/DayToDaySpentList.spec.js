import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import DayToDaySpentList from './DayToDaySpentList';

configure({ adapter: new Adapter() });

describe('DayToDay spent list', () => {
  const data = [{
    category: 'test',
    amount: 1,
  }];
  const sliceColor = [];
  const render = shallow(<DayToDaySpentList data={data} sliceColor={sliceColor} />);

  it('renders as expected', () => {
    expect(render).toMatchSnapshot();
  });
});
