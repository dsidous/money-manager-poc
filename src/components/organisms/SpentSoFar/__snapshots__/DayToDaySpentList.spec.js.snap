// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayToDay spent list renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DayToDaySpentList
    data={
      Array [
        Object {
          "amount": 1,
          "category": "test",
        },
      ]
    }
    sliceColor={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "0",
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <StyledText
            style={
              Object {
                "color": undefined,
                "marginRight": 10,
              }
            }
          >
            ●
          </StyledText>
          <StyledText
            style={
              Object {
                "fontSize": 23,
              }
            }
          >
            test
          </StyledText>
        </Component>,
        <StyledText
          style={
            Object {
              "fontSize": 24,
            }
          }
        >
          £1
        </StyledText>,
      ],
      "style": Object {
        "borderBottomColor": "#ccc",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 20,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <StyledText
              style={
                Object {
                  "color": undefined,
                  "marginRight": 10,
                }
              }
            >
              ●
            </StyledText>,
            <StyledText
              style={
                Object {
                  "fontSize": 23,
                }
              }
            >
              test
            </StyledText>,
          ],
          "style": Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "●",
              "style": Object {
                "color": undefined,
                "marginRight": 10,
              },
            },
            "ref": null,
            "rendered": "●",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "test",
              "style": Object {
                "fontSize": 23,
              },
            },
            "ref": null,
            "rendered": "test",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "£1",
          "style": Object {
            "fontSize": 24,
          },
        },
        "ref": null,
        "rendered": "£1",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "0",
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <StyledText
              style={
                Object {
                  "color": undefined,
                  "marginRight": 10,
                }
              }
            >
              ●
            </StyledText>
            <StyledText
              style={
                Object {
                  "fontSize": 23,
                }
              }
            >
              test
            </StyledText>
          </Component>,
          <StyledText
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            £1
          </StyledText>,
        ],
        "style": Object {
          "borderBottomColor": "#ccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 20,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <StyledText
                style={
                  Object {
                    "color": undefined,
                    "marginRight": 10,
                  }
                }
              >
                ●
              </StyledText>,
              <StyledText
                style={
                  Object {
                    "fontSize": 23,
                  }
                }
              >
                test
              </StyledText>,
            ],
            "style": Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "●",
                "style": Object {
                  "color": undefined,
                  "marginRight": 10,
                },
              },
              "ref": null,
              "rendered": "●",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "test",
                "style": Object {
                  "fontSize": 23,
                },
              },
              "ref": null,
              "rendered": "test",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "£1",
            "style": Object {
              "fontSize": 24,
            },
          },
          "ref": null,
          "rendered": "£1",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
